asdf/_tests/test_api.py

```diff
@@ -119,23 +118,6 @@ def test_atomic_write(tmp_path, small_tree):
         ff.write_to(tmpfile)
 
 
-def test_overwrite(tmp_path):
```

Comment:
The issue mentioned for this test:
https://github.com/asdf-format/asdf/issues/100

describes that it was added to try and replicate an issue on windows. The test did not replicate the issue: https://github.com/asdf-format/asdf/pull/302#issuecomment-320997092
yet the test was left in place.